### PR TITLE
[Concurrency] Split out the default cooperative executor into a separate opt-in library

### PR DESCRIPTION
**Pull request by kubamracek**
Currently, the Embedded Swift's `libswift_Concurrency.a` includes an implementation of a default cooperative executor (via CooperativeGlobalExecutor.cpp) inside, and the mechanism to use a user-provided executor instead relies on the linker preferring code from a .o file and/or order of libraries listed on the linker time. This feels unintuitive and possibly error prone, so this PR changes the setup: Instead, `libswift_Concurrency.a` stops including *any global executor*, and the user is expected to either explicitly link a new library, `libswift_ConcurrencyDefaultExecutor.a`, or link in their own executor. If they don't do either, they'll receive a (pretty self-explanatory) linker error about missing symbols like `swift_task_enqueueGlobalImpl`, etc.

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci Please Build Toolchain macOS Platform

File: stdlib/public/Concurrency/CMakeLists.txt [hunk 1]
```diff
@@ -291,8 +292,37 @@ if(SWIFT_SHOULD_BUILD_EMBEDDED_STDLIB AND SWIFT_SHOULD_BUILD_EMBEDDED_CONCURRENC
...
+    swift_install_in_component(
+      TARGETS embedded-concurrency-default-executor-${mod}
+      DESTINATION "lib/swift/embedded/${mod}"
+      COMPONENT "stdlib"
+      )
+    swift_install_in_component(
+      FILES "${SWIFTLIB_DIR}/embedded/${mod}/libswift_ConcurrencyDefaultExecutor.a"
+      DESTINATION "lib/swift/embedded/${mod}/"
+      COMPONENT "stdlib"
+      PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
```
Comment (al45tair) [inline stdlib/public/Concurrency/CMakeLists.txt:322]: `.a` files don't need or want execute permission. Most systems don't even need it for _dynamic_ libraries, but it's definitely not a thing for static libraries.

**Review by al45tair — APPROVED** [review]

**Comment by kubamracek** [comment]
@swift-ci please test

File: stdlib/public/Concurrency/CMakeLists.txt (hunk 1 above)
Comment (kubamracek) [reply stdlib/public/Concurrency/CMakeLists.txt:322]: Agree in principle, but see <https://github.com/swiftlang/swift/pull/73599> -- there is some pre-existing toolchain packaging logic that requires the +x permission even on .a libraries. Sounds like something worth looking into, I'll try to do that later.

Comment (al45tair) [reply stdlib/public/Concurrency/CMakeLists.txt:322]: It's minor so definitely not needed for this PR. Would be nice to fix that though, as it's pretty weird having the `x` bits set for these.

**Pull request merged**